packages/core/core/src/Asset.js

```diff
@@ -131,22 +256,38 @@ export default class Asset implements IAsset {
     return Array.from(this.dependencies.values());
   }
 
-  createChildAsset(result: TransformerResult) {
-    let code = (result.output && result.output.code) || result.code || '';
-    let opts: AssetOptions = {
-      hash: this.hash || md5FromString(code),
+  createChildAsset(result: TransformerResult): Asset {
+    let content = result.content || result.code || '';
+
+    let hash;
+    let size;
+    if (content === this.content) {
+      hash = this.hash;
+      size = this.stats.size;
+    } else if (typeof content === 'string' || content instanceof Buffer) {
+      hash = md5FromString(content);
+      size = content.length;
+    } else {
+      hash = null;
+      size = NaN;
```

Comment:
Unfortunately we can't know the length of a stream (nor its hash) until we read from it.